packages/domains-table/src/domains-table/domains-table-email-indicator.tsx

```diff
@@ -1,36 +1,85 @@
-import { LoadingPlaceholder } from '@automattic/components';
 import { PartialDomainData } from '@automattic/data-stores';
+import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
-//eslint-disable-next-line no-restricted-imports
-import { useGetMailboxes } from 'calypso/data/emails/use-get-mailboxes';
+import { type as domainTypes } from '../utils/constants';
+import { getDomainType } from '../utils/get-domain-type';
+import { emailManagementEdit } from '../utils/paths';
+
+interface DomainsTableEmailIndicatorProps {
+	domain: PartialDomainData;
+	siteSlug: string;
+}
 
 export const DomainsTableEmailIndicator = ( {
 	domain,
 	siteSlug,
-}: {
-	domain: PartialDomainData;
-	siteSlug?: string;
-} ) => {
-	const { __ } = useI18n();
-	const { data: mailboxes = [], isLoading } = useGetMailboxes( domain.blog_id, {
-		retry: 2,
-	} );
-
-	if ( isLoading || ! siteSlug ) {
-		return <LoadingPlaceholder style={ { width: '50%' } } />;
+}: DomainsTableEmailIndicatorProps ) => {
+	const { __, _n } = useI18n();
+
+	const domainType = getDomainType( domain );
+
+	if ( domainType !== domainTypes.MAPPED && domainType !== domainTypes.REGISTERED ) {
+		return '-';
 	}
 
-	if ( mailboxes.length === 0 ) {
+	const googleStatus = domain.google_apps_subscription?.status;
+	const titanStatus = domain.titan_mail_subscription?.status;
+
+	let message = '';
+
+	if ( googleStatus && ! [ 'no_subscription', 'other_provider' ].includes( googleStatus ) ) {
+		const count = domain.google_apps_subscription?.total_user_count ?? 0;
```

Comment:
Similarly, I would recommend using selectors (such as [this one](https://github.com/Automattic/wp-calypso/blob/47e407de3adf2ee722a1a5de88b3094b632d6abb/client/lib/gsuite/get-gsuite-mailbox-count.js)) instead of accessing properties directly. Note we have selectors for [Google Workspace](https://github.com/Automattic/wp-calypso/tree/47e407de3adf2ee722a1a5de88b3094b632d6abb/client/lib/gsuite), [Titan](https://github.com/Automattic/wp-calypso/tree/7eae3e888d9cdd559c5da041068180debda46b1a/client/lib/titan) as well as [email forwards](https://github.com/Automattic/wp-calypso/tree/09e109d74bfb2af42fa20d2b1a7febf2da3e1bb6/client/lib/domains/email-forwarding) (I'm not going to mention all the places below where we should use them).